ReportCardParser/test/reportcardparser/ReportCardStatsTest.java

```diff
@@ -5,33 +5,129 @@
  */
 package reportcardparser;
 
-import org.junit.jupiter.api.Test;
+import org.junit.Test;
+import static org.junit.Assert.*;
 
 /**
  *
  * @author chris
  */
 public class ReportCardStatsTest {
 
+        IReportCardParser[] reportCardParsers = {
```

Comment:
You have written integration tests, not unit tests.  Your tests rely on the use of your parsing implementations and the text files provided in the project.  What if there is a bug in your parsing implementations?  The input for these unit tests should be ReportCard objects that you construct, without the need for parsing text files.  With this approach, you can cover tests cases like:
- null ReportCard objects
- empty ReportCard arrays
- averages for subjects that don’t necessarily exist in all report card files